### PR TITLE
enhancement(external docs): improve `datadog_agent` source doc

### PR DESCRIPTION
**Pull request by prognant**
This PR should not be merged before the Agent 6/7.35 release is out (as datadog/datadog-agent#10860 has to be included for this PR to be relevant)

### CODE REVIEW

**Comment by binarylogic** [comment]
@prognant these docs look good to me. What's the current status on the agent side?

**Comment by prognant** [comment]
@binarylogic the doc update is ready for release https://github.com/DataDog/documentation/pull/13076 as soon as the the agent 7.35 is release. I updated the PR to reflect the proper version.

File: website/cue/reference/components/sources/datadog_agent.cue [hunk 1]
```diff
@@ -177,16 +177,26 @@ components: sources: datadog_agent: {
 		decompression: {
 			title: "Configuring the Datadog Agent"
```
Comment (spencergilbert) [inline website/cue/reference/components/sources/datadog_agent.cue:178]: I think we ought to split this into two, providing the old and new config for a while

Comment (prognant) [reply website/cue/reference/components/sources/datadog_agent.cue:178]: I don't think there are much users for those options out there (yet), so focusing on the newest configuration seems okay for me, anyway the main point here was to remain consistent with https://docs.datadoghq.com/agent/vector_aggregation/#agent-configuration (BTW I'm only allowed to enable automerge, and doc-only PR never trigger soak tests which are blocking the automatic merge 😞 cc @spencergilbert I think you have the right to merge it)

**Pull request merged**